frontend/dockerfile/dockerfile2llb/convert.go

```diff
@@ -263,16 +263,16 @@ func toDispatchState(ctx context.Context, dt []byte, opt ConvertOpt) (*dispatchS
 	shlex := shell.NewLex(dockerfile.EscapeToken)
 	outline := newOutlineCapture()
 
-	// Validate that base images continue to be valid even
-	// when no build arguments are used.
-	validateBaseImagesWithDefaultArgs(stages, shlex, argCmds, globalArgs, lint)
-
 	// Rebuild the arguments using the provided build arguments
 	// for the remainder of the build.
+	startArgs := globalArgs
 	globalArgs, outline.allArgs, err = buildMetaArgs(globalArgs, shlex, argCmds, opt.BuildArgs)
 	if err != nil {
 		return nil, err
 	}
+	// Validate that base images continue to be valid even
+	// when no build arguments are used.
+	validateBaseImagesWithDefaultArgs(stages, shlex, startArgs, argCmds, lint)
```

Comment:
why was this moved? It doesn't seem to use return values from `buildMetaArgs`

Reply:
Oh, yup! That was moved around while I was refactoring something else. Fixed!